src/App.tsx

```diff
@@ -17,18 +36,36 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                todos={todos}
+                onFilter={setVisibleTodos}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading
+                ? (<Loader />)
+                : (
+                  <TodoList
+                    todos={visibleTodos}
+                    selectedTodoId={selectedTodoId}
+                    onSelect={setSelectedTodoId}
+                  />
+                )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {
+        selectedTodo !== null
```

Comment:
this would be enough
```suggestion
        selectedTodo
```